### ISSUE TITLE
Performance (using hash table with faster hasher)

### ISSUE THREAD

**Issue by s-d-m** [s-d-m/local_jira#8]
### Problem
In order to provide meaningful logs, `Local_jira` computes the changes between the data on the remote and those saved locally from the last synchronisation point.  This allows for providing logs mentioning for example the number of fields of a jira issue which actually changed since the last synchronisation. This feature is helpful for developers wanting to ensure code changes work as expected. For example, if after two immediate synchronisation, the second one still finds changes, chances are that there is a bug in `Local_jira`
and the synchronisation didn't really happen.

The computation of the differences is mostly done using hash sets and then computing the difference to weed out duplicates.  Since it uses hash sets, a hasher function is used extensively. The hasher functions used is the one from the standard library, which was designed to be resistent to DOS attack due to collisions.  The downside is that the hash function is a relatively slow one compared to other hashing functions which do not provide this guarantee. 

In our case, the data that gets hashed are the content of jira tickets, jira comments, etc...  If someone inputting data in a jira database is a realistic threat model in your case, there might be better ways to mitigate the issue, for example setting up a spam filter, banning the user, etc...  In the context of `Local_jira`, I don't believe this threat vector is realistic, and therefore the property of resilience against hash collisions to be necessary. 

In other words, there are some performances to be gained by using a faster hash function such as `fxhash`, or `ahash`.

### Solution
Use a faster hasher function

### Difficulty
Easy-Medium
